src/data-access/repositories/mongo/mongo-action-repository.ts

```diff
@@ -17,14 +19,25 @@ export class MongoActionRepository extends BaseMongoRepository implements Action
   }
 
   public async getAction(actionId: string): Promise<Action> {
+    this.assertDatabaseConnection(this.getAction.name)
     return await this.getCollection().findOne({id: actionId}) as unknown as Action
   }
 
   public async saveActions(actions: Action[]): Promise<void> {
+    this.assertDatabaseConnection(this.saveActions.name)
     await Promise.all(
       actions.map(action =>
         this.getCollection().updateOne({ _id: action.id }, { $set: action }, { upsert: true, ignoreUndefined: true })
       )
     )
   }
+
+  public async deleteActionsForRundown(rundownId: string): Promise<void> {
+    this.assertDatabaseConnection(this.deleteActionsForRundown.name)
+    const actionsDeleteResult: DeleteResult = await this.getCollection().deleteMany({ rundownId: rundownId })
+
+    if (!actionsDeleteResult.acknowledged) {
+      throw new DeletionFailedException(`Failed to delete Actions for Rundown: ${rundownId}`)
```

Comment:
It would be nice if we had consensus on how to structure error and log messages. It is not something that should be done for this PR though :D